test/gtest/ucs/test_rcache.cc

```diff
@@ -287,6 +295,43 @@ UCS_MT_TEST_F(test_rcache, get_unmapped, 6) {
     free(ptr);
 }
 
+UCS_MT_TEST_F(test_rcache, put_and_invalidate, 1) {
+    static const size_t size = 1 * 1024 * 1024;
+    std::vector<region*> regions;
+    void *ptr;
+    size_t i;
+    size_t j;
+    size_t comp_count;
+
+    ptr = malloc(size);
+    for (i = 1; i < 100; i++) {
+        comp_count   = (i + 1) / 2;
+        m_comp_count = 0;
+        for (j = 0; j < i; j++) {
+            regions.push_back(get(ptr, size));
+        }
+    
+        for (j = 0; j < i; j++) {
+            /* mix region put and invalidate operations on same region */
+            ASSERT_EQ(0, m_comp_count);
+            region *region = regions.back();
+            if ((j & 1) == 0) { /* on even iteration invalidate region */
```

Comment:
use rand?

Reply:
> we have to set personal seed for this test

yes, we have a seed for the whole gtest

Reply:
the problem is in reproducibility: set of test cases will depends from tests sequences/test filters/etc... it will be hard to reproduce issue